src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -1184,3 +1192,498 @@ class DMFT(ModelMethodElectronic):
+                f'{self.approximation_type} normally approximates exchange as well; '
+                f"but approximated_term='{self.approximated_term}'."
+            )
+
+
+class PerturbationMethod(ModelMethodElectronic):
+    type = Quantity(
+        type=MEnum('MP', 'RS', 'BW'),
+        default='MP',
+        description="""
+        Perturbation approach. The abbreviations stand for:
+        | Abbreviation | Description |
+        | ------------ | ----------- |
+        | `'MP'`       | Møller-Plesset |
+        | `'RS'`       | Rayleigh-Schrödigner |
+        | `'BW'`       | Brillouin-Wigner |
+        """,
+    )
+
+    order = Quantity(
+        type=np.int32,
+        description="""
+        Order up to which the perturbation is expanded.
+        """,
+    )
+
+    density = Quantity(
+        type=MEnum('relaxed', 'unrelaxed'),
+        description="""
+        unrelaxed density: MP2 expectation value density.
+        relaxed density  : incorporates orbital relaxation.
+        """,
+    )
+
+    spin_component_scaling = Quantity(
+        type=MEnum('none', 'SCS', 'SOS', 'custom'),
+        default='none',
+        description="""
+        Spin-component scaling approach for perturbation methods:
+          - none  : no spin-component scaling
+          - SCS   : spin-component scaled (Grimme's approach, https://doi.org/10.1002/wcms.1110)
+          - SOS   : spin-opposite scaled
+          - custom: user-defined scaling factors
+        This is typically relevant only for MP2 calculations.
+        """,
+    )
+
+
+class LocalCorrelation(BaseModelMethod):
+    """
+    A base section used to define the parameters of a local correlation method for
+    post-HF calculations, e.g. LMP2, LCC, or domain-based local pair natural orbitals
+    (PNO, LPNO, DLPNO) in coupled cluster or double-hybrid DFT.
+
+    Typical references:
+      - Pulay, Chem. Phys. Lett. 100, 151 (1983) (LMP2 concept).
+      - G. Knizia, G. K.-L. Chan, “Density Matrix Embedding,” J. Chem. Theory Comput. 9, 1428 (2013).
+      - F. Neese, “The ORCA program system,” WIREs Comput. Mol. Sci. 2, 73-78 (2012)
+        for DLPNO approaches.
+    """
+
+    type = Quantity(
+        type=MEnum('PNO', 'LPNO', 'DLPNO', 'LMP2', 'other'),
+        description="""
+        The local correlation flavor:
+          - 'PNO'   : Pair Natural Orbitals in generic form
+          - 'LPNO'  : Local PNO approach
+          - 'DLPNO' : Domain-based Local PNO approach
+          - 'LMP2'  : Local MP2 (Pulay approach)
+          - 'other' : Another local correlation scheme
+        """,
+    )
+
+    parent_correlation_method = Quantity(
+        type=MEnum('MP2', 'CC', 'DH-DFT'),
+        description="""
+        The underlying electron-correlation ansatz upon which the local treatment is built:
+      - 'MP2'      : local MP2
+      - 'CC'       : local CC
+      - 'DHDFT'    : local double-hybrid DFT
+        """,
+    )
+
+
+class MolecularModelMethod(ModelMethodElectronic):
```

Comment:
what is the convention or class names GeneralSpecfic or the other way? should this be ElectronicModelMethod? or ModelMethodMolecular?